### PR TITLE
Add no-op support in iree/testing/benchmark library

### PR DESCRIPTION
**Pull request by hcindyl**
Support no-op functions when external benchmark library is not available (currently turned off when `-DIREE_ENABLE_THREADING=OFF`)

### CODE REVIEW

File: iree/testing/benchmark_nops.c [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// Create no-ops for the functions in benchmark.h
```
Comment (benvanik) [inline iree/testing/benchmark_nops.c:7]: remove

File: iree/testing/CMakeLists.txt [hunk 2]
```diff
@@ -1,28 +1,41 @@
...
+    PUBLIC
+  )
+else()
+  iree_cc_library(
+    NAME
+      benchmark
+    HDRS
+      "benchmark.h"
+    SRCS
+      "benchmark_nops.c"
```
Comment (benvanik) [inline iree/testing/CMakeLists.txt:32]: _nop (for consistency with iree/task/task_test_nop)

**Review by benvanik — CHANGES_REQUESTED** [review]

File: iree/testing/benchmark_nops.c (hunk 1 above)
Comment (hcindyl) [reply iree/testing/benchmark_nops.c]: Done.

File: iree/testing/CMakeLists.txt (hunk 2 above)
Comment (hcindyl) [reply iree/testing/CMakeLists.txt]: Done.

**Review by benvanik — APPROVED** [review]

**Pull request merged**